viewer/packages/pointclouds/src/potree-three-loader/tree/PointCloudOctreePickerHelper.ts

```diff
@@ -0,0 +1,351 @@
+import {
+  BufferAttribute,
+  Camera,
+  Color,
+  LinearFilter,
+  NearestFilter,
+  NoBlending,
+  Points,
+  Ray,
+  RGBAFormat,
+  Scene,
+  Sphere,
+  Vector3,
+  Vector4,
+  WebGLRenderer,
+  WebGLRenderTarget
+} from 'three';
+import { COLOR_BLACK } from '../rendering/constants';
+import { ClipMode, PointCloudMaterial, PotreePointColorType } from '../rendering';
+import { PointCloudOctree } from './PointCloudOctree';
+import { IPointCloudTreeNode } from './IPointCloudTreeNode';
+import { PickPoint, PointCloudHit } from '../types/types';
+
+export interface RenderedNode {
+  node: IPointCloudTreeNode;
+  octree: PointCloudOctree;
+}
+
+export interface IPickState {
+  renderTarget: WebGLRenderTarget;
+  material: PointCloudMaterial;
+  scene: Scene;
+}
+
+export interface PickParams {
+  pickWindowSize: number;
+  pickOutsideClipRegion: boolean;
+  /**
+   * If provided, the picking will use this pixel position instead of the `Ray` passed to the `pick`
+   * method.
+   */
+  pixelPosition: Vector3;
+  /**
+   * Function which gets called after a picking material has been created and setup and before the
+   * point cloud is rendered into the picking render target. This gives applications a chance to
+   * customize the renderTarget and the material.
+   *
+   * @param material The pick material.
+   * @param renterTarget The render target used for picking.
+   */
+  onBeforePickRender: (material: PointCloudMaterial, renterTarget: WebGLRenderTarget) => void;
+}
+
+/**
+ * Helper class for PointCloudOctreePicker.
+ */
+export class PointCloudOctreePickerHelper {
+  private static readonly helperVec3 = new Vector3();
+  private static readonly helperSphere = new Sphere();
+  private static readonly clearColor = new Color();
+
+  public static prepareRender(
```

Comment:
This seems a bit dodgy because it changes state, but I dont really find a guarantee that its restored

Reply:
I'd prefer if we were using `WebGLRendererStateHelper` and extended it to support the fields necessary here.